games/devtest/mods/experimental/commands.lua

```diff
@@ -43,6 +43,79 @@ minetest.register_chatcommand("test_bulk_set_node", {
 	end,
 })
 
+-- Safeguard against too much optimization.
+_G._bench_content_ids_data = setmetatable({}, {__mode = "k"})
+
+local function bench_name2content()
+	local t = {}
+	_G._bench_content_ids_data[t] = true
+
+	local get_content_id = minetest.get_content_id
+
+	local start = minetest.get_us_time()
+
+	for i = 1, 200 do
+		for name in pairs(minetest.registered_nodes) do
+			t[#t + 1] = get_content_id(name)
+		end
+	end
+
+	local finish = minetest.get_us_time()
+
+	return (finish - start) / 1000
+end
+
+local function bench_content2name()
+	local t = {}
+	_G._bench_content_ids_data[t] = true
+
+	local n = 0
+	for _ in pairs(minetest.registered_nodes) do
+		n = n + 1
+	end
+
+	local get_name_from_content_id = minetest.get_name_from_content_id
+	local CONTENT_UNKNOWN = minetest.CONTENT_UNKNOWN
+
+	local start = minetest.get_us_time()
+
+	for i = 1, 200 do
+		for j = 0, n do
+			if j ~= CONTENT_UNKNOWN then
```

Comment:
why?

Reply:
I forgot that `CONTENT_UNKNOWN` is cached too, so it won't throw off the optimization. I'll change this.